rethinkdb/net.py

```diff
@@ -48,7 +46,7 @@
 from rethinkdb.handshake import HandshakeV1_0
 from rethinkdb.logger import default_logger
 
-__all__ = ['connect', 'set_loop_type', 'Connection', 'Cursor', 'DEFAULT_PORT']
```

Comment:
This is abusing `__all__` because of the setattr happening in `RethinkDB` init. Otherwise this is going to override the `connect` method defined in the class on instance instantiations. Without `connect` here Python rules implicitly say it's a private method, then we expect it above. I'm not a fan of the confusion in all that. Renaming to `make_connection` or changing it to a class would solve this.